### PR TITLE
refactor: remove `LetRec`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to https://github.com/flix/flix/issues/8782

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala [hunk 1]
```diff
@@ -280,6 +278,8 @@ object SyntaxTree {
 
       case object LiteralVector extends Expr
 
+      case object LocalDef extends Expr
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala:281]: Why is this here now? 😕

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala:281]: What I mean is: I thought this was _already here_

**Review by magnus-madsen — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala (hunk 1 above)
Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala:281]: No I just reused the letrec parsing and then renamed it (in this pr)... this is where it should be alphabetically

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala:281]: OK
